src/main/java/org/cryptomator/cryptofs/health/dirid/OrphanDir.java

```diff
@@ -34,4 +68,104 @@ public String toString() {
+		} else {
+			String uuid = Files.readString(cipherRecoveryDirFile, StandardCharsets.UTF_8);
+			if (!uuid.equals(Constants.RECOVERY_DIR_ID)) {
+				throw new FileAlreadyExistsException("Directory /" + Constants.RECOVERY_DIR_NAME + " already exists, but with wrong directory id.");
+			}
+		}
+		String recoveryDirHash = cryptor.hashDirectoryId(Constants.RECOVERY_DIR_ID);
+		Path cipherRecoveryDir = dataDir.resolve(recoveryDirHash.substring(0, 2)).resolve(recoveryDirHash.substring(2)).toAbsolutePath();
+		Files.createDirectories(cipherRecoveryDir);
+
+		//create deorphanedDirectory
+		String cipherStepParentDirName = convertClearToCiphertext(cryptor, clearStepParentDirName, Constants.RECOVERY_DIR_ID);
+		Path cipherStepParentDirFile = cipherRecoveryDir.resolve(cipherStepParentDirName + "/" + Constants.DIR_FILE_NAME);
+		Files.createDirectory(cipherStepParentDirFile.getParent());
+		var stepParentUUID = UUID.randomUUID().toString();
+		Files.writeString(cipherStepParentDirFile, stepParentUUID, StandardCharsets.UTF_8, StandardOpenOption.CREATE_NEW, StandardOpenOption.WRITE);
+		String stepParentDirHash = cryptor.hashDirectoryId(stepParentUUID);
+		Path stepParentDir = dataDir.resolve(stepParentDirHash.substring(0, 2)).resolve(stepParentDirHash.substring(2)).toAbsolutePath();
+		Files.createDirectories(stepParentDir);
+		return new CryptoPathMapper.CiphertextDirectory(stepParentUUID, stepParentDir);
+	}
+
+	// visible for testing
+	void adoptOrphanedResource(Adoption adoption, CryptoPathMapper.CiphertextDirectory stepParentDir, FileNameCryptor cryptor, String longNameSuffix) throws IOException {
+		if (adoption.oldCipherPath.toString().endsWith(Constants.DEFLATED_FILE_SUFFIX)) {
+			var newCipherName = convertClearToCiphertext(cryptor, adoption.newClearname + longNameSuffix, stepParentDir.dirId);
+			var deflatedName = BaseEncoding.base64Url().encode(SHA1_HASHER.digest(newCipherName.getBytes(StandardCharsets.UTF_8))) + Constants.DEFLATED_FILE_SUFFIX;
+			Path targetPath = stepParentDir.path.resolve(deflatedName);
+			Files.move(adoption.oldCipherPath, targetPath, StandardCopyOption.ATOMIC_MOVE);
+
+			//adjust name.c9s
+			try (var fc = Files.newByteChannel(targetPath.resolve(Constants.INFLATED_FILE_NAME), StandardOpenOption.WRITE, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+				fc.write(ByteBuffer.wrap(newCipherName.getBytes(StandardCharsets.UTF_8)));
+			}
+		} else {
+			var newCipherName = convertClearToCiphertext(cryptor, adoption.newClearname, stepParentDir.dirId);
+			Path targetPath = stepParentDir.path.resolve(newCipherName);
+			Files.move(adoption.oldCipherPath, targetPath, StandardCopyOption.ATOMIC_MOVE);
+		}
+	}
+
+	// visible for testing
+	record Adoption(String newClearname, Path oldCipherPath) {}
+
+	private String createClearnameToBeShortened(int threshold) {
+		int neededLength = threshold / 4 * 3 - 16;
```

Comment:
I forgot we need to subtract `.c9r` → 4 chars from the ciphertext name as well. See also:

https://github.com/cryptomator/cryptofs/blob/6445f1ddb06c201dfa6e4e9c81459201c17dec70/src/main/java/org/cryptomator/cryptofs/common/FileSystemCapabilityChecker.java#L99-L101

Reply:
Fixed in 6af26a6